.github/workflows/build-linux-musl.yml

```diff
@@ -0,0 +1,89 @@
+name: Build for linux-musl
+
+on:
+  workflow_call:
+  workflow_dispatch:
+
+jobs:
+  build:
+    name: Build
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - arch: x64
+            platform: linux/amd64
+          - arch: ia32
+            platform: linux/386
+          - arch: arm64
+            platform: linux/arm64
+          - arch: arm
+            platform: linux/arm/v7
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Compile Protobuf
+        run: |
+          docker run --rm -i \
+                     --volume "$PWD:$PWD" \
+                     --workdir "$PWD" \
+                     docker.io/library/dart <<'EOF'
+          set -e
+          curl -fsSL -H "Authorization: Bearer ${{ github.token }}" "https://github.com/bufbuild/buf/releases/latest/download/buf-$(uname -s)-$(uname -m).tar.gz" | tar -xzC /usr/local --strip-components 1
+          dart pub get
+          dart run grinder protobuf
+          EOF
+
+      # There is a bug in qemu's mremap causing pthread_getattr_np in musl to stuck in a loop on arm.
+      # Unless qemu fixes the bug or we get a real linux-arm runner, we cannot build aot-snapshot
+      # for arm on CI. So, we create an kernel snapshot for arm build in amd64 container instead.
+      #
+      # In addition, we need to rename the artifact from -linux- to -linux-musl- to avoid conflict
+      # with glibc builds.
+      - name: Build
+        run: |
+          docker run --rm -i \
+                     --platform ${{ matrix.arch == 'arm' && 'linux/amd64' || matrix.platform }} \
+                     --volume "$PWD:$PWD" \
+                     --workdir "$PWD" \
+                     ghcr.io/dart-musl/dart <<'EOF'
+          set -e
+          dart pub get
+          dart run grinder pkg-standalone-linux-${{ matrix.arch }}
+          find build -name '*.tar.gz' -print0 | xargs -0 -n 1 -- sh -xc 'mv "$1" "$(echo "$1" | sed -e "s/linux/linux-musl/")"' --
+          EOF
+
+      # The kernel snapshot created for arm in the previous step is bundling a glibc based dart runtime
+      # due to how cli_pkg download the sdk for building non-native platforms. Therefore we need to
+      # replace it with musl-libc based dart runtime to create a working linux-musl-arm package.
```

Comment:
Would it make sense to build more of this logic into `cli-pkg`? We could teach it `pkg-standalone-linux-musl-*` with your SDK releases.

Reply:
That's certainly possible, but I would consider that at a later time due to the complexity. The problem mainly lies on dart-sdk only have the knowledge of itself running as some "linux" variant, and have no concept of "linux-gnu" or "linux-musl", which would require lots of structural changes to cli_pkg to accommodate this. Also, we need a way to know what is the current running variant, that either the current running dart-sdk need to be able to tell what kind of linux variant by itself, or we need to parse the dart executable file as ELF, and then detect what it is based on the interpreter field.

Reply:
Would it be possible for your fork to add "musl" somewhere in the `dart --version` output? That seems like the easiest way to tell—we don't need an actual API call in the SDK itself necessarily.

Reply:
My "fork" does not modify any dart-sdk source code. Only thing it does is running a CI/CD with a different build toolchain for building dart-sdk on alpine linux.

The reason I said it is difficult to do in either dart or cli_pkg is because the platform logic in dart is closely tied to the `Abi` class in `ffi` package. For example, the `dart --version` pretty much prints [`Abi.current.toString()`](https://api.dart.dev/stable/3.2.3/dart-ffi/Abi/toString.html). In order to print musl, we can possibly either add new constant, but this will likely break existing code that does the platform detection via `Abi` class; or modify `dart --version` to hard code a different string instead of using constant, which again might be unexpected.

I don't think change the version string is the correct behavior. I think the right way to deal with this in dart-sdk would be to provide new API that returns the ["target triplet"](https://wiki.osdev.org/Target_Triplet). For example, in ruby we have `RbConfig::CONFIG['arch']` which would return `x86_64-linux-musl` that clearly identify the right platform.